tests/calico_cni_k8s_test.go

```diff
@@ -2004,12 +2142,18 @@ var _ = Describe("Kubernetes CNI tests", func() {
 				"projectcalico.org/orchestrator":   api.OrchestratorKubernetes,
 				"projectcalico.org/serviceaccount": "default",
 			}))
+
 			endpointSpec = endpoints.Items[0].Spec
-			Expect(endpointSpec.ContainerID).Should(Equal(containerID))
+			if os.Getenv("DATASTORE_TYPE") != "kubernetes" {
+				Expect(endpointSpec.ContainerID).Should(Equal(containerID))
+			}
 			checkIPAMReservation()
 		})
 
 		AfterEach(func() {
+			// Delete pod
```

Comment:
I think this actually breaks the testcase right below with this error:

```
  Expected error:

      <*types.Error | 0xc000f4bd40>: {

          Code: 100,

          Msg: "pods \"run601964353\" not found",

          Details: "",

      }

      pods "run601964353" not found

  not to have occurred
```

Reply:
I did not reproduce your issue here. But I will adjust the order for AfterEach:
```
		AfterEach(func() {
			_, err = testutils.DeleteContainerWithId(netconf, contNs.Path(), name, testutils.K8S_TEST_NS, containerID)
			Expect(err).ShouldNot(HaveOccurred())

			ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name)
		})
```